src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -166,111 +169,135 @@ public async Task<List<InstanceTypeInfo>> ListOfAvailableInstanceTypes()
+                    LoadBalancerArns = new List<string>
+                    {
+                        loadBalancerArn
+                    }
+                });
+
+                if (!loadBalancers.LoadBalancers.Any())
+                {
+                    throw new AWSResourceNotFoundException(DeployToolErrorCode.LoadBalancerDoesNotExist, $"The load balancer '{loadBalancerArn}' does not exist.");
+                }
 
-            return loadBalancers.LoadBalancers.First();
+                return loadBalancers.LoadBalancers.First();
+            });
         }
 
         public async Task<List<Amazon.ElasticLoadBalancingV2.Model.Listener>> DescribeElasticLoadBalancerListeners(string loadBalancerArn)
         {
             var elasticLoadBalancingClient = _awsClientFactory.GetAWSClient<IAmazonElasticLoadBalancingV2>();
-
-            var listeners = await elasticLoadBalancingClient.DescribeListenersAsync(new DescribeListenersRequest
+            return await HandleException(async () =>
             {
-                LoadBalancerArn = loadBalancerArn
-            });
+                var listeners = await elasticLoadBalancingClient.DescribeListenersAsync(new DescribeListenersRequest
+                {
+                    LoadBalancerArn = loadBalancerArn
+                });
 
-            if (!listeners.Listeners.Any())
-            {
-                throw new AWSResourceNotFoundException(DeployToolErrorCode.LoadBalancerListenerDoesNotExist, $"The load balancer '{loadBalancerArn}' does not have any listeners.");
-            }
+                if (!listeners.Listeners.Any())
+                {
+                    throw new AWSResourceNotFoundException(DeployToolErrorCode.LoadBalancerListenerDoesNotExist, $"The load balancer '{loadBalancerArn}' does not have any listeners.");
+                }
 
-            return listeners.Listeners;
+                return listeners.Listeners;
+            });
         }
 
         public async Task<DescribeRuleResponse> DescribeCloudWatchRule(string ruleName)
         {
             var cloudWatchEventsClient = _awsClientFactory.GetAWSClient<IAmazonCloudWatchEvents>();
-
-            var rule = await cloudWatchEventsClient.DescribeRuleAsync(new DescribeRuleRequest
+            return await HandleException(async () =>
             {
-                Name = ruleName
-            });
+                var rule = await cloudWatchEventsClient.DescribeRuleAsync(new DescribeRuleRequest
+                {
+                    Name = ruleName
+                });
 
-            if (rule == null)
-            {
-                throw new AWSResourceNotFoundException(DeployToolErrorCode.CloudWatchRuleDoesNotExist, $"The CloudWatch rule'{ruleName}' does not exist.");
-            }
+                if (rule == null)
+                {
+                    throw new AWSResourceNotFoundException(DeployToolErrorCode.CloudWatchRuleDoesNotExist, $"The CloudWatch rule'{ruleName}' does not exist.");
```

Comment:
For a case like this one where we have an existing error code for "does not exist" - **are we loosing granularity via the new exception handling?**

Per [the docs](https://docs.aws.amazon.com/eventbridge/latest/APIReference/API_DescribeRule.html#API_DescribeRule_Errors) it looks like this will throw a 400 - ResourceNotFoundException in this case.

Are we now going to throw with `DeployToolErrorCode.ResourceQuery` instead of the `DeployToolErrorCode.CloudWatchRuleDoesNotExist`?

(didn't run this code, so let me know if I'm misunderstanding the flow)

Reply:
We indeed wrap that exception but on the screen customer will see a well formatted error message like I have in description.

In the case you mentioned, it should be something like when no diagnostics option passed 
```
 ResourceNotFoundException
 An entity that you specified does not exist.
```

Reply:
I was also thinking about this from the perspective of the optional telemetry we get back from the VS Toolkit that lets us chase down common bugs or sharp edges.

Do we still get whats in the `MessageBuilder` back?

Reply:
Confirmed with @philasmar, Toolkit will report ErrorCode and Message provided to the `DeployToolException`

In this particular we are overriding the error which means original thrown from the service is lost and we will have a custom error message.